### ISSUE TITLE
[Question]: How can I use axe-core as a functional step to provide accessibility remarks on design system components?

### ISSUE THREAD

**Issue by phun-ky** [dequelabs/axe-core#2845]
This could be more of a SO question, but the documentation didn't get me anywhere regarding my wanted usage, so I put it here.

Basically, this is what I want: 

Whenever the design system I create builds, axe-core should be run on predefined html-templates for each component to check accessibility. The outcome of that test should be reflected in the design system documentation, stating if a component is fully accessible and safe to use in production.

So:

* npm run build 
  * preprocessor files are build
  * documentation is updated (if some markdown files are generated)
    * Design tokens
    * Classnames
    * Accessibility remarks <-- i want axe results here
* publish 

From what I can tell, you basically need to kickstart  a webserver and serve a static html file with the desired html templates to test, get the results, kill the webserver, generated documentation based on results. 

Does this sound legit? (This is not react components)

**Comment by straker** [comment]
I guess that depends on what the components are. If they are just static HTML (and not using Raect or anything) you could use jsdom to turn the string file into DOM elements that axe can work with. You can see a working example of that here https://github.com/dequelabs/axe-core/blob/develop/doc/examples/jsdom/test/a11y.js.

If that doesn't work then I think you would have to start of a server and a webpage and run something like [@axe-core/webdriverio](https://www.npmjs.com/package/@axe-core/webdriverio) to get results you can work with.

**Comment by phun-ky** [comment]
I created a working POC for this now: https://gist.github.com/phun-ky/e2de34d7c5f17b004520e6928b8281f0#file-a11y-component-badge-js-L117

At line 117, I get the results I needed. But it is a bit spaghetti code...I read somewhere in the docs that using jsdom did not give the most correct results, or has that changed?

**Comment by straker** [comment]
Since jsdom is javascript implementation of the browser renderer it could have some troubles with accurate results.

**Comment by dylanb** [comment]
Please move this discussion to the axe community Slack
